src/components/DigitalCredentials/CredentialsStepper.vue

```diff
@@ -0,0 +1,237 @@
+                href=""
+                target="_blank"
+              >
+                I didn't receive anything
+              </a>
+            </v-card-text>
+            <v-card-text
+              v-else
+              class="d-flex flex-column"
+            >
+              <div
+                class="d-flex justify-self-center align-self-center justify-center align-center"
+                style="width: 200px; height: 200px"
+              >
+                <v-icon
+                  color="black"
+                  size="xxx-large"
+                >
+                  mdi-check-circle-outline
+                </v-icon>
+              </div>
+              <p class="justify-center text-center font-weight-bold word-wrap">
+                Your Business Card is now ready to use
+              </p>
+            </v-card-text>
+          </v-card>
+        </v-col>
+      </v-row>
+    </v-container>
+    <CredentialsWebSocket
+      @onConnection="handleConnection"
+      @onIssuedCredential="handleIssuedCredential"
+    />
+  </section>
+</template>
+
+<script lang="ts">
+import { DigitalCredentialTypes, Routes } from '@/enums'
+import { LegalServices } from '@/services'
+import { useBusinessStore } from '@/stores'
+import { Getter } from 'pinia-class'
+import { Component, Vue } from 'vue-property-decorator'
+import QrcodeVue from 'qrcode.vue'
+import { DigitalCredentialIF, WalletConnectionIF } from '@/interfaces'
+import CredentialsWebSocket from '@/components/DigitalCredentials/CredentialsWebSocket.vue'
+
+Component.registerHooks(['beforeRouteEnter'])
+
+// Create a component that extends the Vue class called CredentialsStepper
+@Component({ components: { QrcodeVue, CredentialsWebSocket } })
+export default class CredentialsStepper extends Vue {
+  @Getter(useBusinessStore) getIdentifier!: string;
+
+  credentialTypes = DigitalCredentialTypes;
+  credentialConnection: WalletConnectionIF = null;
+  issuedCredential: DigitalCredentialIF = null;
+
+  async beforeRouteEnter (to, from, next): Promise<void> {
+    next(async (_this) => {
+      const { data } = await LegalServices.fetchCredentials(_this.getIdentifier)
+      const issuedCredentials = data?.issuedCredentials
+
+      if (issuedCredentials?.length) {
+        next({ name: Routes.DIGITAL_CREDENTIALS })
+      } else {
+        next()
+      }
+    })
+  }
+
+  async mounted (): Promise<void> {
+    await this.getCredentialsConnection()
+    if (!this.credentialConnection) {
+      await this.addCredentialInvitation()
+    }
+  }
+
+  async addCredentialInvitation (): Promise<void> {
+    const { data: connection } = await LegalServices.createCredentialInvitation(this.getIdentifier)
+    this.credentialConnection = connection
+  }
+
+  async getCredentialsConnection (): Promise<void> {
+    const { data } = await LegalServices.fetchCredentialConnections(this.getIdentifier)
+    this.credentialConnection = data?.connections?.[0]
```

Comment:
Conditional chaining can return undefined, which might make this variable un-responsive. I recommend a fallback value (eg, null).

Reply:
Looks good.